frontend/components/Timer.js

```diff
@@ -1,74 +1,160 @@
+        }));
+    }
+
+    const onError = (error) => {
+        console.log("Error: " + error)
+    }
+
+    const onMessageReceived = (payload) => {
+        let millis = JSON.parse(payload.body).time
+        let minutes = Math.floor(millis / 60000)
+        let seconds = Number(((millis % 60000) / 1000).toFixed(0))
+        setMinutes(minutes)
+        setSeconds(seconds)
+    }
+
     const toggle = () => {
+        if(!isActive) {
+            sendStart()
+        } else {
+            sendStop()
+        }
         setIsActive(!isActive)
     }
 
     const reset = () => {
-        sendTimeToDatabase('-end')
-        setMinutes(0)
-        setSeconds(0)
-        setIsActive(false)
+        if(isActive){
+            setIsActive(!isActive)
+        }
+        sendReset()
+        return Alert.alert(
+            "Are your sure?",
+            "Are you sure you want to reset the timer?",
+            [
+                {
+                    text: "Yes",
+                    onPress: () => {
+                        sendReset()
+                    },
+                },
+                {
+                    text: "No",
+                },
+            ]
+        );
     }
 
-    const sendTimeToDatabase = (param) => {
-        Api.post(`/timer${param}/token`, {
-            minutes: minutes,
-            seconds: seconds
-        }).then(res => {
-            console.log(res)
-        }).catch(e => {
-            console.log(e)
-        })
+    const end = () => {
+        sendEnd()
+        return Alert.alert(
+            "Are your sure?",
+            "Are you sure you want to end the match?",
+            [
+                {
+                    text: "Yes",
+                    onPress: () => {
+                        sendEnd()
+                    },
+                },
+                {
+                    text: "No",
+                },
+            ]
+        );
     }
 
-    useEffect(() => {
-        let interval = null
-        if(isActive){
-            interval = setInterval(() => {
-                if(seconds === 59){
-                    setMinutes(minutes + 1)
-                    setSeconds(0)
-                } else{
-                    setSeconds(seconds + 1)
-                }
-                sendTimeToDatabase('')
-            }, 1000)
-        } else if (!isActive && (seconds !== 0 && minutes !== 0)){
-            clearInterval(interval)
+    const sendStart = () => {
+        if(stompClient) {
+            let message = {
+                action: "START"
+            };
+            stompClient.send('/timer', {}, JSON.stringify(message));
+        }
+    }
+
+    const sendStop = () => {
+        if(stompClient) {
+            let message = {
+                action: "STOP"
+            };
+            stompClient.send('/timer', {}, JSON.stringify(message));
+        }
+    }
+
+    const sendReset = () => {
+        if(stompClient) {
+            let message = {
+                action: "RESET"
```

Comment:
```suggestion
                action: "RESTART"
```
Back has this one in enum, should working now

Reply:
i changed this to RESET in backend, but I didnt add it in this branch, it is added to my branch with tests.

Reply:
when i did git add . i was in frontend folder and it didnt add backend folders, that's why it is what it is

Reply:
So i would leave it as it is now, as soon as we will merge my branch with tests it will be working.